src/app/shared/services/openWeatherMaps.js

```diff
@@ -0,0 +1,52 @@
+export default class OpenWeatherMaps {
+    constructor($http, fiveDayForecast, latestError, currentForecast) {
+        this.$http = $http;
+        this.fiveDayForecast = fiveDayForecast;
+        this.latestError = latestError;
+        this.currentForecast = currentForecast;
+    }
+
+    getCurrentForecastCity(city) {
+        const URL = 'https://api.openweathermap.org/data/2.5/weather';
+
+        const params = {
+            params: {
+                q: city,
+                units: 'metric',
+                lang: 'en',
+                appid: process.env.OPENWEATHERMAP_API_KEY
+            }
+        };
+
+        return this.$http.get(URL, params)
+            .then((response) => {
+                this.currentForecast = response;
+            })
+            .catch((response) => {
+                this.latestError = response;
+            });
+    };
+
+    getFiveDayForecastCity(city) {
+        const URL = 'https://api.openweathermap.org/data/2.5/forecast';
```

Comment:
You should create a baseApi with 'https://api.openweathermap.org/data/2.5/' and build each endpoint url by concatenating the baseApi with the specific part